fission/src/mirabuf/MirabufInstance.ts

```diff
@@ -126,39 +125,35 @@ class MirabufInstance {
     }
 
     /**
-     * Parses all mirabuf appearances into ThreeJs materials.
+     * Parses all mirabuf appearances into ThreeJS and Jolt materials.
      */
     private LoadMaterials(materialStyle: MaterialStyle) {
+        Object.entries(this._mirabufParser.assembly.data!.materials!.physicalMaterials!).forEach(([name, material]) => {
+            const [static_f, dynamic_f] = [material.staticFriction, material.dynamicFriction]
+            console.log(`${name} - static: ${static_f} dynamic: ${dynamic_f}`)
+            //this._materials.set()
+        })
         Object.entries(this._mirabufParser.assembly.data!.materials!.appearances!).forEach(
             ([appearanceId, appearance]) => {
-                let hex = 0xe32b50
-                let opacity = 1.0
-                if (appearance.albedo) {
-                    const { A, B, G, R } = appearance.albedo
-                    if (A && B && G && R) {
-                        hex = (A << 24) | (R << 16) | (G << 8) | B
-                        opacity = A / 255.0
-                    }
-                }
-
-                let material: THREE.Material
-                if (materialStyle == MaterialStyle.Regular) {
-                    material = new THREE.MeshPhongMaterial({
-                        color: hex,
-                        shininess: 0.0,
-                        shadowSide: THREE.DoubleSide,
-                        opacity: opacity,
-                        transparent: opacity < 1.0,
-                    })
-                } else if (materialStyle == MaterialStyle.Normals) {
-                    material = new THREE.MeshNormalMaterial()
-                } else if (materialStyle == MaterialStyle.Toon) {
-                    material = World.SceneRenderer.CreateToonMaterial(hex, 5)
-                    console.debug("Toon Material")
-                }
-
-                World.SceneRenderer.SetupMaterial(material!)
-                this._materials.set(appearanceId, material!)
+                const { A, B, G, R } = appearance.albedo ?? {}
+                const [hex, opacity] =
+                    A && B && G && R ? [(A << 24) | (R << 16) | (G << 8) | B, A / 255.0] : [0xe32b50, 1.0]
+
+                const material =
+                    materialStyle === MaterialStyle.Regular
+                        ? new THREE.MeshPhongMaterial({
+                              color: hex,
+                              shininess: 0.0,
+                              shadowSide: THREE.DoubleSide,
+                              opacity: opacity,
+                              transparent: opacity < 1.0,
+                          })
+                        : materialStyle === MaterialStyle.Normals
+                          ? new THREE.MeshNormalMaterial()
+                          : World.SceneRenderer.CreateToonMaterial(hex, 5)
```

Comment:
I'm generally in favor of using ternaries whenever possible but this, at least from my perspective, is certainly an aggressive use of them. Although I do think that this is fine to leave, I suppose I'm just not used to seeing this doubly nested ternary syntax.

Reply:
I personally find it easier to read and understand the intention of whoever wrote it, but we can ask around to decide on a standard

Reply:
I definitely think it's fine to leave. Now that you mentioned the inconsistent formatting that is also bugging me. Something we can revisit later, totally fine for now.